builder/googlecompute/config.go

```diff
@@ -178,6 +178,9 @@ type Config struct {
 	//  }
 	//  ```
 	ImageStorageLocations []string `mapstructure:"image_storage_locations" required:"false"`
+	// The name of the disk that should be used to create an image. Defaults to the
+	// value of the `disk_name` setting if not set.
+	ImageSourceDisk string `mapstructure:"image_source_disk" required:"false"`
```

Comment:
Come to think of it, this only makes sense if one of the extra disks is to become the image source, and it's supposedly one only.
Should this be a flag on one of the extra disks? Something like `is_image_source` maybe?

Basically this amounts to what you've added as a private flag to the structure, but instead making it public and settable in the configs.

Then we'd need to check if we have this flag set for one disk, and propagate the created disk as the target for the image creation step.

What do you think? Let me know if the approach sounds good to you, and if you need a hand getting this in, I can help, just say the word :)